src/elements/content-sidebar/SidebarContentInsights.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import noop from 'lodash/noop';
+import { FormattedMessage } from 'react-intl';
+
+import ContentInsightsSummary from '../../features/content-insights/ContentInsightsSummary';
+import { GraphData } from '../../features/content-insights/types';
+// @ts-ignore Module is written in Flow
+import messages from '../common/messages';
+// @ts-ignore Module is written in Flow
+import SidebarSection from './SidebarSection';
+// @ts-ignore Module is written in Flow
+// eslint-disable-next-line import/no-named-as-default, import/no-named-as-default-member
+import withErrorHandling from './withErrorHandling';
+
+interface ContentInsights {
+    error?: Object;
+    graphData: GraphData;
+    isLoading: boolean;
+    previousPeriodCount: number;
+    totalCount: number;
+}
+
+interface Props {
+    contentInsights?: ContentInsights;
+    error?: Object;
+    onContentInsightsClick?: () => void;
+    onContentInsightsMount?: () => void;
+}
+
+const defaultContentInsights = {
+    graphData: [],
+    isLoading: true,
+    previousPeriodCount: 0,
+    totalCount: 0,
+};
+
+const SidebarContentInsights = ({
+    contentInsights = defaultContentInsights,
+    onContentInsightsClick = noop,
+    onContentInsightsMount = noop,
+}: Props) => {
+    const [hasMounted, setHasMounted] = React.useState(false);
+    const { graphData, isLoading, previousPeriodCount, totalCount } = contentInsights;
+
+    React.useEffect(() => {
+        if (hasMounted) {
+            return;
+        }
+
+        onContentInsightsMount();
```

Comment:
What was the reasoning for exposing the internal lifecycle here, again?

Reply:
So that the Content Insights summary in the details sidebar can initiate a call to fetch the data if none has yet been cached

Reply:
Got it, thanks. Should this be handled in `DetailsSidebar` like it is for access stats? It also supports a `refresh` method, that does something similar on-demand.

Reply:
Sure, let me explore this option